source/fundamentals/auth.txt

```diff
@@ -575,3 +576,146 @@ mechanism:
 For additional information on configuring your application to use
 certificates as well as TLS/SSL options, see our
 :doc:`TLS/SSL guide </fundamentals/connection/tls>`.
+
+``MONGODB-OIDC``
+~~~~~~~~~~~~~~~~
+
+.. important::
+
+   The MONGODB-OIDC authentication mechanism requires MongoDB v7.0 or later running
+   on a Linux platform.
+
+The following sections describe how to use the MONGODB-OIDC authentication mechanism to
+authenticate to various platforms.
+
+For more information about the MONGODB-OIDC authentication mechanism, see
+:manual:`OpenID Connect Authentication </core/security-oidc/>` in the MongoDB Server
+manual.
+
+.. _java-mongodb-oidc-azure-imds:
+
+Azure IMDS
+++++++++++
+
+If your application runs on an Azure VM, or otherwise uses the
+`Azure Instance Metadata Service <https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service>`__
+(IMDS), you can authenticate to MongoDB by using the {+driver-short+}'s built-in Azure
+support.
+
+You can specify Azure IMDS OIDC authentication on a ``MongoClientSettings`` object either by 
+using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax.
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-azure-imds-connection-string
+    
+      The following code example shows how to specify Azure IMDS OIDC authenticaton.
+      Replace the ``<resource>`` placeholder with the percent-encoded application or
+      service that the OIDC access token is intended for.
+     
+      .. code-block:: java
+         
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>@<hostname>:<port>/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:<resource>");
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-azure-mongo-credential
+
+      The following code example shows how to specify Azure IMDS OIDC authenticaton.
+      Replace the ``<username>`` placeholder with the client ID of the Azure managed
+      identity, and replace the ``<resource>`` placeholder with the percent-encoded
+      application or service that the OIDC access token is intended for.
+      
+      .. code-block:: java
+
+         MongoCredential credential = MongoCredential.createOidcCredential("<username>")
+            .withMechanismProperty("ALLOWED_HOSTS", allowedHosts)
```

Comment:
I couldn't information on this property anywhere, beyond it being a `List<String>`; I'm guessing it's new as part of the OIDC work. It would be good if we could provide some info on it.

Reply:
It is new, and important, but is only used when a human callback is being used.

Reply:
What do we want users to know about it?

Reply:
found in auth spec